rapids_dask_dependency/dask_loader.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import importlib
+import importlib.abc
+import importlib.machinery
+import sys
+from contextlib import contextmanager
+
+from .patches.dask import patches as dask_patches
+from .patches.distributed import patches as distributed_patches
+
+
+class DaskLoader(importlib.abc.MetaPathFinder, importlib.abc.Loader):
+    def create_module(self, spec):
+        if spec.name in ("dask", "distributed"):
+            with self.disable():
+                mod = importlib.import_module(spec.name)
+            # Note: The spec does not make it clear whether we're guaranteed that spec
+            # is not a copy of the original spec, but that is the case for now. We need
+            # to assign this because the spec is used to update module attributes after
+            # it is initialized by create_module.
+            spec.origin = mod.__spec__.origin
+            spec.submodule_search_locations = mod.__spec__.submodule_search_locations
+            return mod
```

Comment:
Okay cool - So, would `create_module` be the place to add hooks for patching?

For example, to silence the `dask.dataframe` deprecation warning, could we do something like

```python
mod.config.set({"dataframe.query-planning-warning": False})
```

before returning?

Reply:
Right now I've put that in `exec_module` (that's what the loop over patches is), but on second thought I think we can probably put all the logic in `create_module`. The way I've structured it now we have a list of idempotent patch functions to be called.

Reply:
I've changed this in the latest, so all patching can happen in `create_module`. We should add in some real patches in place of my samples and see how those look.